src/nvim/undo.c

```diff
@@ -2321,7 +2322,8 @@ static void u_undoredo(int undo)
   if (undo)
     /* We are below the previous undo.  However, to make ":earlier 1s"
      * work we compute this as being just above the just undone change. */
-    --curbuf->b_u_seq_cur;
+    curbuf->b_u_seq_cur =  curhead->uh_next.ptr ?
+        curhead->uh_next.ptr->uh_seq : 0;
```

Comment:
The comment above this is now a bit outdated, would be good to give some brief account of the explanation from the PR description.